hanadb_exporter/exporters/prometheus_exporter.py

```diff
@@ -62,20 +70,22 @@ def collect(self):
         execute db queries defined by metrics_config/api file, and store them in
         a prometheus metric_object, which will be served over http for scraping e.g gauge, etc.
         """
-
         for query in self._metrics_config.queries:
             if not query.enabled:
                 self._logger.info('Query %s is disabled', query.query)
             elif not utils.check_hana_range(self._hana_version, query.hana_version_range):
                 self._logger.info('Query %s out of the provided hana version range: %s',
                                   query.query, query.hana_version_range)
             else:
-                # TODO: manage query error in an exception
-                query_result = self._hdb_connector.query(query.query)
-                formatted_query_result = utils.format_query_result(query_result)
-                for metric in query.metrics:
-                    if metric.type == "gauge":
-                        metric_obj = self._manage_gauge(metric, formatted_query_result)
-                        yield metric_obj
-                    else:
-                        raise NotImplementedError('{} type not implemented'.format(metric.type))
+                try:
+                    query_result = self._hdb_connector.query(query.query)
+                    formatted_query_result = utils.format_query_result(query_result)
+                    for metric in query.metrics:
+                        if metric.type == "gauge":
+                            metric_obj = self._manage_gauge(metric, formatted_query_result)
+                            yield metric_obj
+                        else:
+                            raise NotImplementedError('{} type not implemented'.format(metric.type))
+                except hdb_connector.connectors.base_connector.QueryError as err:
+                    self._logger.logger.error('Failure in query: %s, skipping...', query.query)
```

Comment:
I think that will be misleading. Sometimes the difference between queries is a column name for example that can be missed if we truncate the log ...

Reply:
I agree with both of you, so maybe we should log the file line number or some kind of query ID?

Reply:
That will be better actually!

Reply:
I support the idea of creating some king of a query ID or name. This will require having an extra attribute in the JSON file, if you are ok I can create a card for it.